### PR TITLE
Add producer to write out pt and decay mode dependent tau id scale factors

### PR DESCRIPTION
**Pull request by mburkart**

### CODE REVIEW

File: interface/HttSettings.h [hunk 1]
```diff
@@ -392,6 +392,11 @@ class HttSettings: public KappaSettings {
 	IMPL_SETTING_STRINGLIST_DEFAULT(TauTrigger2017IDTypes, {});
 	IMPL_SETTING_STRINGLIST_DEFAULT(TauTrigger2017EfficiencyWeightNames, {});
 
+	// settings for TauIDEfficiencyProducer
```
Comment (ArturAkh) [inline interface/HttSettings.h:395]: The name is TauIDScaleFactorProduce, right? Please change the comment accordingly

File: python/data/ArtusConfigs/Run2LegacyAnalysis/2016/et.py [hunk 2]
```diff
@@ -268,7 +283,7 @@ def build_config(nickname, **kwargs):
   if not isData and not isEmbedded:                 config["Processors"].append( "producer:RooWorkspaceWeightProducer")
   if isEmbedded:                 config["Processors"].append( "producer:EmbeddedWeightProducer")
   if isEmbedded:                 config["Processors"].append( "producer:TauDecayModeWeightProducer")
-  #if not isData:                 config["Processors"].append( "producer:TauTrigger2017EfficiencyProducer")
```
Comment (ArturAkh) [inline python/data/ArtusConfigs/Run2LegacyAnalysis/2016/et.py:271]: Why is TauTrigger2017EfficiencyProducer not applied for 2016 in et, mt & tt? Do we have a different setup there?

Comment (mburkart) [reply python/data/ArtusConfigs/Run2LegacyAnalysis/2016/et.py:271]: As far as I know we had a different setup so far. I think we will be switching to a more consistent setup when including the embedding tau trigger efficiencies.

Comment (ArturAkh) [reply python/data/ArtusConfigs/Run2LegacyAnalysis/2016/et.py:271]: ok fine with me

File: interface/HttSettings.h (hunk 1 above)
Comment (mburkart) [reply interface/HttSettings.h:395]: Done.

**Pull request merged**